lib/hydrothermalVenture.ex

```diff
@@ -0,0 +1,42 @@
+defmodule AdventCode2 do
+  def createRange([{a, b}, {a, c}]) do
+    Enum.to_list(b..c)
+    |> Enum.map(fn x -> {a, x} end)
+  end
+
+  def createRange([{a, b}, {c, b}]) do
+    Enum.to_list(a..c)
+    |> Enum.map(fn x -> {x, b} end)
+  end
+
+  def run(path) do
+    variants = path
+    |> File.stream!()
+    |> Stream.map(&String.trim/1)
+    |> Stream.map(&String.split(&1, " -> ", trim: true))
+    |> Stream.map(fn [a, b] ->
+      [x1, y1] = String.split(a, ",")
+      [x2, y2] = String.split(b, ",")
+
+      [
+        {String.to_integer(x1), String.to_integer(y1)},
+        {String.to_integer(x2), String.to_integer(y2)}
+      ]
+    end)
```

Comment:
Я пошел по пути наименьшего сопротивления: выдрал цифры.
```suggestion
    |> Stream.map(&Regex.scan(~r/\d+/, &1))
    |> Stream.map(fn [[x1], [y1], [x2], [y2]] -> 
      [
        {String.to_integer(x1), String.to_integer(y1)},
        {String.to_integer(x2), String.to_integer(y2)}
      ]
     end)
```